statistics/scalar.go

```diff
@@ -70,9 +71,28 @@ func convertDatumToScalar(value *types.Datum, commonPfxLen int) float64 {
 		return float64(value.GetMysqlDuration().Duration)
 	case types.KindMysqlTime:
 		valueTime := value.GetMysqlTime()
-		zeroTime := types.ZeroDatetime
-		zeroTime.Type = valueTime.Type
-		return float64(valueTime.Sub(&zeroTime).Duration)
+		var minTime types.Time
+		switch valueTime.Type {
+		case mysql.TypeDate:
+			minTime = types.Time{
+				Time: types.MinDatetime,
+				Type: mysql.TypeDate,
+				Fsp:  types.DefaultFsp,
+			}
+		case mysql.TypeDatetime:
+			minTime = types.Time{
+				Time: types.MinDatetime,
+				Type: mysql.TypeDatetime,
+				Fsp:  types.DefaultFsp,
+			}
+		case mysql.TypeTimestamp:
+			minTime = types.Time{
+				Time: types.MinTimestamp,
+				Type: mysql.TypeTimestamp,
+				Fsp:  types.DefaultFsp,
+			}
+		}
+		return float64(valueTime.Sub(&minTime).Duration)
```

Comment:
if valueTime.isZero(), then it will log error again.

Reply:
This will not happen because this function will be called only when the value is greater than the bucket boundaries, and the bucket boundaries will at least as large as `minTime`.